packages/compile-solidity/compilerSupplier/loadingStrategies/LoadingStrategy.js

```diff
@@ -78,22 +78,31 @@ class LoadingStrategy {
   }
 
   markListeners() {
-    return new Set(process.listeners("uncaughtException"));
+    return {
+      uncaughtException: new Set(process.listeners("uncaughtException")),
+      unhandledRejection: new Set(process.listeners("unhandledRejection")),
+    };
   }
 
   /**
    * Cleans up error listeners left by older versions of soljson (pre-0.5.1).
    * Use with `markListeners()`
    */
   removeListener(markedListeners) {
-    const listeners = process.listeners("uncaughtException");
-    const newListeners = listeners.filter(
-      listener => !markedListeners.has(listener)
-    );
+    const scrub = (eventName) => {
+      const listeners = process.listeners(eventName);
+      const newListeners = listeners.filter(
+        (listener) => !markedListeners[eventName].has(listener),
+      );
+
+      for (const listener of newListeners) {
+        console.log("removing ...listener");
+        process.removeListener(eventName, listener);
+      }
+    };
 
-    for (const listener of newListeners) {
-      process.removeListener("uncaughtException", listener);
-    }
+    scrub("uncaughtException");
+    scrub("unhandledRejection");
```

Comment:
part of me thinks this `scrub()` is just extra overhead for no good reason. why not just double the for loop?

```javascript
for (const eventName of ["uncaughtException", "unhandledRejection"]) {
  /* ... */
}
```

this might be more readable?

Reply:
Love readability!